text/0002-searchable-display-attributes-behaviors.md

```diff
@@ -0,0 +1,116 @@
+### Summary
+
+We wanted to change `searchable_attributes`, beacause it was confusing. We recently add wildcard in this field to select all. This last change introduced lot of edge-cases:
+- we don't know the order of searchable attributes when we use wildcards.
+
+Searchable attributes have several uses:
+- know in which fields we can search.
+- know the importance of fields for the criterion `Attributes`.
+
+When a wildcard is used, the order of searchable attributes is written no-where.
+
+Is it relevant to mix this 2 information when we have wildcards?
+### Explanation
+
+#### default values
+
+`searchable_attributes`: *   
+`displayed_attributes`: *
+
+#### wildcard (*) & `searchable_attributes` order
+
+> When the searchable attributes value is set to ["*"] the priority of the documents attributes is undefined.
+
+By default searchable attributes should be in the order of the first indexed document, for example:
+```json
+{
+  "title": "many the fish",
+  "description": "it is awfull to be liked by many"
+}
+```
+default `searchable_attributes` order should be `["title", "description"]` where `"title" > "description"`
+
+When a wildcard is set we should have the default behavior.
+
+#### `searchable_attributes` & `displayed_attributes`
+
+what is the behavior for a field that:
+- > **is in `searchable_attributes` and in `displayed_attributes`:**
+  > meilisearch will use the field for a search and the field will be returned in search results
+- > **is in `searchable_attributes` but not in `displayed_attributes`:**
+  > meilisearch will use the field for a search but the field will **not** be returned in search results
+- > **is not in `searchable_attributes` but in `displayed_attributes`:**
+  > meilisearch will **not** use the field for a search but the field will be returned in search results
+- > **is not in `searchable_attributes` and not in `displayed_attributes`:**
+  > meilisearch will **not** use the field for a search and the field will **not** be returned in search results. The field is unused but should be kept in case of it is added in `searchable_attributes` or `displayed_attributes`.
+
+#### add a new configuration `attributes_order`?
+
+This new config may define the order of attributes for the criterion `Attributes`, no wildcard allowed.
+if `searchable_attribute` is: `["title", "desc"]`
+and `attributes_order` is: `["desc", "id", "title"]`
+The searchable attributes order would be:
+`"desc" > "title"` (`"id"` is not searchable so is ignored by the criterion `Attributes`).
+
+`attributes_order` is always exhaustive, If a new field is added, it is "pushed" at the end of the list.
+
+##### drawbacks:
+- new config added
+- more complexity
+- could be replace by a small note about wildcard in `searchable_attributes`
+
+#### Changes in meilisearch documentation
+
+Depending on what we choose, we'll have to add **default** and **wildcard** behavior of configurations, we'll potentially add a new configuration `attributes_order`. 
```

Comment:
I would like us to consider the possibility of removing the wildcard, as it seems "simple" for usage at first sight, but introduces complexity as behavior is not clear and needs to be documented/learned, and is subject to confusion.

This also would make it possible to use just one setting, stating CLEARLY that the order MATTERS! So it could solve both issues without adding a new setting!

Reply:
This is not an option. We decided to add the wildcard because without it we needed acceptNewField whose use needed an understanding of how meilisearch used new fields and was not clear at all

Reply:
The more I reflect on this, the more I feel the need to create `attributes_order` as in this proposition. I would try hard to avoid adding a new setting, but it seems to be the only way (for now) to offer logic, consistent and clear behavior for this issue.

Reply:
> I think that MeiliSearch being easy to use is more about the fact that you have very little configuration to do to make it work in most case than the fact that we have "few settings".

I think it is a mix of both

Reply:
I thought a lot about this, and I also feel really concerned about the implicit-and-not-obvious role of `searchableAttributes`, i.e the importance order of the fields.

And in this case, adding another settings might be the best option to manage the importance of the fields during the search.

Let's call this new setting `XXX`.

Here what we should be aware about if we add this new settings:
- Adding a new setting is adding an obstacle for the user if there are too many settings, even if each settings are explicit and well documentated. I had bad experience the first time I tried Algolia and I saw all the possibilities. No need to detail my experience with Elastic even if all the Elastic settings are really accurate and the docs not that bad. So, yes, adding a settings is adding complexity from my POV. We should be carefull about settings addition, but it does not mean not adding another settings at all. If `XXX` is needed, we should add it.
- The name of `XXX` should be really accurate
- We should anticipate, at the maximum, all the behaviors between `searchableAttributes` and `XXX`. What's about if there are less fields in `searchableAttributes` than in `XXX`?
- We should anticipate all the behavior possibilities when the users try to update `XXX`: if the list of attributes is not exhaustive, does MeiliSearch return an error or do an implicit behavior?
- How the `XXX` is going to be filled if the users add documents with new fields?

If we don't add `XXX` the docs should be really well explained and the users should be able to understand as soon as possible the `searchableAttributes` settings allow them to do two things.

The more I'm thinking about this, the more I would like to add a new setting, but I don't have any strong position about this. And if we add this new settings, we should be careful about a lot of behaviors.

Reply:
And we definitely need to talk directly about this to take a decision: adding a new settings or not 🙂

Reply:
Seems like time has come for a meeting! :)